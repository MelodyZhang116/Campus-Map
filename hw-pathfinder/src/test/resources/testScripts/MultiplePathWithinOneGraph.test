# test that one graph with multiple path finding. Those paths will not interact with each other.

CreateGraph complexGraph

AddNode complexGraph A
AddNode complexGraph B
AddNode complexGraph C
AddNode complexGraph D
AddNode complexGraph E

AddEdge complexGraph A B 1.0
AddEdge complexGraph B E 100.0
AddEdge complexGraph C D 50.0
AddEdge complexGraph D E 1.0
AddEdge complexGraph A C 20.0
AddEdge complexGraph B A 75.8333333333
AddEdge complexGraph B C 13.13133
AddEdge complexGraph B D 57.1

FindPath complexGraph A E
FindPath complexGraph B E
FindPath complexGraph A E
