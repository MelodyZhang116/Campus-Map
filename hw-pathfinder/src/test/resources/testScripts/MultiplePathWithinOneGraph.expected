# test that one graph with multiple path finding. Those paths will not interact with each other.

created graph complexGraph

added node A to complexGraph
added node B to complexGraph
added node C to complexGraph
added node D to complexGraph
added node E to complexGraph

added edge 1.000 from A to B in complexGraph
added edge 100.000 from B to E in complexGraph
added edge 50.000 from C to D in complexGraph
added edge 1.000 from D to E in complexGraph
added edge 20.000 from A to C in complexGraph
added edge 75.833 from B to A in complexGraph
added edge 13.131 from B to C in complexGraph
added edge 57.100 from B to D in complexGraph

path from A to E:
A to B with weight 1.000
B to D with weight 57.100
D to E with weight 1.000
total cost: 59.100
path from B to E:
B to D with weight 57.100
D to E with weight 1.000
total cost: 58.100
path from A to E:
A to B with weight 1.000
B to D with weight 57.100
D to E with weight 1.000
total cost: 59.100